### ISSUE TITLE
Freezing, Dropout ve Weighted Loss

### ISSUE THREAD

**Issue by MesuttSahin** [MesuttSahin/teknofest_xray_sim#47]
###  BAĞIMLILIK DURUMU
* ** Girdi:** Analistten gelen `pos_weight`.
* ** Çıktı:** Güncellenmiş `model.py` ve `train.py`.

###  Görev Özeti
Modelin kapasitesini kısıtla (Freezing), ezberini boz (Dropout) ve Analistin verdiği ağırlıkları Loss fonksiyonuna ekle.

###  TEKNİK GÖREVLER 

**A. Dosya:** `src/models/model.py`
1.  **Dondurma (Freezing):**
    * Model yüklendikten sonra:
      ```python
      for name, param in self.model.named_parameters():
          if "layer4" not in name and "fc" not in name:
              param.requires_grad = False
      ```
2.  **Dropout Ekleme:**
    * FC katmanını güncelle:
      ```python
      self.model.fc = nn.Sequential(
          nn.Dropout(0.5), # %50 unutma
          nn.Linear(num_ftrs, num_classes)
      )
      ```

**B. Dosya:** `src/train.py` (Eğitim Ayarları)
1.  **Weighted Loss:**
    * `weights = torch.tensor([Analistin_Verdiği_Liste]).to(device)`
    * `criterion = nn.BCEWithLogitsLoss(pos_weight=weights)`
2.  **Optimizer & Scheduler:**
    * `optimizer = torch.optim.AdamW(..., weight_decay=1e-4)`
    * `scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer, patience=2, factor=0.1)`
3.  **Döngü Güncellemesi:**
    * Eğitim döngüsünde epoch sonunda: `scheduler.step(val_loss)` ekle.

###  Checklist
- [ ] `layer4` hariç katmanlar dondu mu?
- [ ] Loss fonksiyonuna `pos_weight` eklendi mi?
